pkg/core/storelimit/sliding_window.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storelimit
+
+import "github.com/tikv/pd/pkg/utils/syncutil"
+
+const (
+	// minSnapSize is the min value to check the windows has enough size.
+	minSnapSize = 10
+)
+
+// SlidingWindows is a multi sliding windows
+type SlidingWindows struct {
+	mu      syncutil.RWMutex
+	windows []*window
+}
+
+// NewSlidingWindows is the construct of SlidingWindows.
+func NewSlidingWindows(cap float64) *SlidingWindows {
+	if cap < 0 {
+		cap = minSnapSize
+	}
+	windows := make([]*window, priorityLevelLen)
+	for i := 0; i < int(priorityLevelLen); i++ {
+		windows[i] = newWindow(int64(cap) >> i)
+	}
+	return &SlidingWindows{
+		windows: windows,
+	}
+}
+
+// Reset resets the capacity of the sliding windows.
+func (s *SlidingWindows) Reset(cap float64, _ Type) {
```

Comment:
I prefer `ResetCapacity` than `Reset`. `Reset` sounds like resetting used, count and capacity.

Reply:
The function belongs the interface of `StoreLimit`. In the `RateLimit` implements, the reset means that the `RateLimit ` will change it's producer speed.